lib/utils/parse/parse.go

```diff
@@ -69,6 +69,34 @@ func (emailLocalTransformer) transform(in string) (string, error) {
 	return parts[0], nil
 }
 
+// regexpReplaceTransformer replaces all matches of re with replacement
+type regexpReplaceTransformer struct {
+	re          *regexp.Regexp
+	replacement string
+}
+
+// newRegexpReplaceTransformer attempts to create a regexpReplaceTransformer or
+// fails with error if the expression does not compile
+func newRegexpReplaceTransformer(expression, replacement string) (*regexpReplaceTransformer, error) {
+	re, err := regexp.Compile(expression)
+	if err != nil {
+		return nil, trace.BadParameter("failed parsing regexp %q: %v", expression, err)
+	}
+	return &regexpReplaceTransformer{
+		re:          re,
+		replacement: replacement,
+	}, nil
+}
+
+// transform applies the regexp replacement (with expansion)
+func (r regexpReplaceTransformer) transform(in string) (string, error) {
+	// filter out inputs which to not match the regexp at all
+	if !r.re.MatchString(in) {
+		return "", nil
```

Comment:
Shouldn't this return `in` unmodified?

Reply:
I'm filtering out inputs which do not match the regex, that seems to be more useful and the behaviour people actually want. Empty strings are removed from the output set in `Interpolate`.